### PR TITLE
Bugfix/coupled crow

### PR DESCRIPTION
**Pull request by jkuang-exploro**
The purpose of this PR is to fix a bug in workflow layout so that standalone atmospheric experiments could be generated properly. It is NOT designed to solve every problem so it's normal if it cannot run from end to end.

I would just want to make sure that it could perform a coupled experiment, as same as the current feature/coupled-crow (trunk). Currently that means run coupled experiment through forecast step.

### CODE REVIEW

File: workflow/defaults/settings.yaml [hunk 1]
```diff
@@ -44,7 +44,13 @@ default_settings: &default_settings
 
   cplflx: .false.
   cplice: !calc cplflx
-  cpl: .false.
+  cpl: !FirstTrue
+    - when: !calc doc.settings.cplflx==".true."
+      do: .true.
+    - when: !calc doc.settings.cplice==".true."
```
Comment (JessicaMeixner-NOAA) [inline workflow/defaults/settings.yaml:53]: - when: !calc doc.settings.cplwav==".true."

cpl should also be true if cplwav is true and cplflx is false.  Same for cplchem

Comment (jkuang-exploro) [reply workflow/defaults/settings.yaml:53]: @JessicaMeixner-NOAA The updating of cplwav is part of a systematic change that I've put together in PR#81. In other words, your suggestion is to essentially merge this one with PR#81 ( which you suggest we should discard?). This PR is branched from the trunk (feature/coupled-crow) which doesn't even have wave components in the configuration system yet. Please let me know what I should do.

**Comment by jiandewang** [comment]
I had one successful run for fcst job this morning, will repeat one more time to see if "hdf5 library error" will appear again or not.

**Comment by jkuang-exploro** [comment]
@jiandewang Thanks! I have a test on my side as well, and going through medcold now.

**Comment by jiandewang** [comment]
@JianKuang-NOAA  cron doesn't work for me, how did you manage that?

**Comment by JianKuang-NOAA** [comment]
@jiandewang You would need to manually edit the crontab file in a text editor and substitute the "rocotorun" with the full path (return of "which rocoto")

**Comment by jkuang-exploro** [comment]
@jiandewang You would need to manually edit the crontab file in a text editor and substitute the "rocotorun" with the full path (return of "which rocotorun")

**Comment by jiandewang** [comment]
@JianKuang-NOAA 
Orion-login-1[72] jiwang$ which rocotorun
rocotorun is hashed (/apps/contrib/rocoto/1.3.1/bin/rocotorun)
so this is what is inside cron
*/5 * * * * /apps/rocoto/1.3.1/bin/rocotorun -d /work/noaa/marine/Jiande.Wang/CROW/wrk/EXPROOT/test_3d/workflow.db -w /work/noaa/marine/Jiande.Wang/CROW/wrk/EXPROOT/test_3d/workflow.xml

but it doesn't work

**Comment by jkuang-exploro** [comment]
@jiandewang You need to put "/apps/contrib/rocoto/1.3.1/bin/rocotorun" instead of "/apps/rocoto/1.3.1/bin/rocotorun"

**Comment by jiandewang** [comment]
@JianKuang-NOAA 
I saw the issue: I need to add contrib here

**Comment by jiandewang** [comment]
@JianKuang-NOAA 
when I do rocotoview, I got
jiwang$ ./rocoto_viewer.py -w workflow.xml -d workflow.db
  File "./rocoto_viewer.py", line 253
    print str(err)
            ^
SyntaxError: invalid syntax

this is my module load:
module load intelpython3

**Comment by bingfu-NOAA** [comment]
> @JianKuang-NOAA
> when I do rocotoview, I got
> jiwang$ ./rocoto_viewer.py -w workflow.xml -d workflow.db
> File "./rocoto_viewer.py", line 253
> print str(err)
> ^
> SyntaxError: invalid syntax
> 
> this is my module load:
> module load intelpython3

@jiandewang 
I used module load intelpython3/2020
it works for me.

**Comment by jiandewang** [comment]
@JianKuang-NOAA 
just switch to intelpython3/2020
still the same thing:
intelpython3/2020
Orion-login-1[96] jiwang$ ./rocoto_viewer.py -w workflow.xml -d workflow.db
  File "./rocoto_viewer.py", line 253
    print str(err)
            ^

**Comment by jiandewang** [comment]
I had one successful fcst job this morning, 
however when I tried to repeat the run, I got
/work/noaa/marine/Jiande.Wang/CROW/PR101/jobs/JGLOBAL_FORECAST_MEDCOLD[44]: setpdy.sh: not found [No such file or directory]

**Comment by JessicaMeixner-NOAA** [comment]
This is probably the lmod->lmod/lmod issue. 

The wave post still does not work but otherwise, the updates with waves (and fixes for lmod -> lmod/lmod are ready for testing.  I could make a new PR for that, and then we could re-test these changes?  Or -- I could add it to your PR 81 @JianKuang-NOAA  if you would prefer that

**Comment by KateFriedman-NOAA** [comment]
An error with setpdy.sh usually means the prod_util module didn't load. Do
you know of the lmod change on Orion? You have to initialize modules with a
different path now (extra "/lmod"): /apps/lmod/lmod/init/$__ms_shell

Kate Friedman (formerly Howard)
NOAA/NWS/NCEP/EMC Engineering and Implementation Branch

**Comment by jiandewang** [comment]
@KateFriedman-NOAA  and @JessicaMeixner-NOAA thanks for the input. But my jobs were running totally fine this morning, post finished ~600H, then around 10:30AM, all post started to failed from 606H and after because of setpdy.sh. So seems to me it's a system issue.

**Comment by JessicaMeixner-NOAA** [comment]
@jiandewang same issue that you are describing to me happened to me as well.  The lmod ->lmod/lmod is what fixed it. I don't know why it works some and not others, but I am fairly sure this will fix it for you.

**Comment by jiandewang** [comment]
I had at least 3 tries where fcst finished properly, though in other cases job failed with either library or setpdy,sh issue, but I suspect they are related to the system rather than the script

**Review by jiandewang — APPROVED** [review]

**Comment by jkuang-exploro** [comment]
@jiandewang Are you approving this PR? You've approved some changes but not the entire PR.

**Comment by jiandewang** [comment]
yes I am approving this PR

**Review by jiandewang — APPROVED** [review]

**Comment by jkuang-exploro** [comment]
@JessicaMeixner-NOAA @KateFriedman-NOAA Jiande Wang has approved this PR but it is still unable to merge? Is that because he is not within a subgroup that was authorized to approve a PR?

**Comment by jkuang-exploro** [comment]
@jiandewang @JessicaMeixner-NOAA I have merged the updated feature/coupled-crow (trunk) into this PR. Please test it and review. Thanks

**Comment by jiandewang** [comment]
@JianKuang-NOAA my testing on Orion: fcst and atmos post ran fine, no setpdy or hdf5 library issue. Could you confirm with me that the success of  these two steps are the purpose of your PR? If yes then I will approve it.

**Comment by jkuang-exploro** [comment]
@jiandewang That's correct. Please approve this PR.

**Review by jiandewang — APPROVED** [review]

**Comment by jkuang-exploro** [comment]
@JessicaMeixner-NOAA Would you please approve this PR?

**Review by JessicaMeixner-NOAA — APPROVED** [review]
cplwav and cplchem were added.  I did not test, but Jiande gave the okay for it running.

**Pull request merged**